crates/bevy_render/src/render_resource/mod.rs

```diff
@@ -44,6 +44,9 @@ pub use wgpu::{
     VertexStepMode,
 };
 
-pub use bevy_crevice::*;
+pub mod encase {
+    pub use bevy_encase_derive::ShaderType;
```

Comment:
I'd prefer to export ShaderType (both the derive and the trait) directly at the render_resource level. This will be a relatively common derive, the longer "nested" path is regrettable, and the context of encase isn't something most bevy users should be thinking about.

I'd also prefer it if we renamed this module from encase to shader_type, given that there will almost certainly be cases where users import those types directly. From the perspective of bevy users, using the encase name is less helpful. I understand that this is a form of debranding, but I think we could solve the issue of "visible credit" in a way that doesn't affect UX (ex: adding encase to the bevy readme like we do for our other core libs. the readme is almost certainly more visible than a niche import). I'd like to hash out this point before merging, but I won't block this PR on making that work (as I believe doing that would require adjusting the encase derive impl).

Reply:
Or maybe alternatively, do the full encase re-export (under the encase name), but then at the render_resource level export everything that will cover normal bevy use cases. This also gives us full control over the namespace / gives us the ability to type alias stuff without stomping on naming assumptions baked into encase.  

I've been experimenting with moving the ShaderType export "out" of the blanket encase export, and sadly if you do this theres a "conflict" error:

```rust
// in render_resource/mod.rs

pub use bevy_encase_derive::ShaderType;
pub use encase::ShaderType;
```

This can be fixed by either:
1. Moving the blanket export up one level (merging it into render_resource). This doesn't seem tenable though because of present and future type conflicts (ex our DynamicUniformBuffer conflicts with encase's). 
2. Remove encase's derive export. Obviously this _also_ isn't tenable because encase needs to be viable as a standalone library.
3. Add an optional feature flag to disable encase's derive export.

None of those are ideal. (3) feels like a reasonable compromise, but thats solely @teoxoy's call.

Reply:
The following seems to work

```rust
pub use self::encase::{ShaderType, Size as ShaderSize};
```

I think having the full encase re-export and then re-exporting the `ShaderType` and `ShaderSize` from it at the root of `render_resource` is a good approach since it provides the flexibilty of having the full re-export and also the ease of use for `ShaderType` and `ShaderSize`.

I'll push a commit with this change - let me know what you think!

Reply:
Arg wait nevermind my bad. I was seeing a different error message! The examples do need to be updated to use the new paths, but everything appears to be fine. I'll push the tweaks.

Reply:
Ah, yeah I somehow missed those. Thanks!